src/diffusers/pipeline_utils.py

```diff
@@ -231,7 +231,7 @@ def device(self) -> torch.device:
             module = getattr(self, name)
             if isinstance(module, torch.nn.Module):
                 if module.device == torch.device("meta"):
-                    return torch.device("cpu")
+                    return torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
@patrickvonplaten @piEsposito this feels hacky, but is required to make the pipelines work when `self.device` is not the same as e.g. `generator` after offloading.
See the error here: https://github.com/huggingface/diffusers/actions/runs/3410777950/jobs/5674151054#step:10:551

Seems that `accelerate` doesn't populate `param_original_devices` [here](https://github.com/huggingface/accelerate/blob/102cf00ded3b1fce4a205c76a98f3645d98a8192/src/accelerate/hooks.py#L229), so the only way to know where the model was supposed to go is to guess?

Reply:
I understand - and actually think this is a clever solution, I've learned a few thing with this PR of yours.

Also, IMO it is correct to assume that, is a user has a GPU, then they will use it instead of CPU for Diffusion models.

Reply:
Actually the more I think about it, wouldn't the cleanest solution be just to return `torch.device("meta")` and then fix the bugs in the pipelines directly. 

Bit worried about making such a fundamental function this hacky.

Also cc @pcuenca - curious to hear your thoughts!